lib/modules/manager/pub/artifacts.spec.ts

```diff
@@ -0,0 +1,227 @@
+      fs.getSiblingFileName.mockReturnValueOnce(lockFile);
+      fs.readLocalFile.mockResolvedValueOnce(oldLockFileContent);
+      fs.readLocalFile.mockResolvedValueOnce(newLockFileContent);
+      expect(
+        await pub.updateArtifacts({
+          ...updateArtifact,
+          newPackageFileContent: params.packageFileContent,
+        })
+      ).toEqual([
+        {
+          file: {
+            type: 'addition',
+            path: lockFile,
+            contents: newLockFileContent,
+          },
+        },
+      ]);
+      expect(execSnapshots).toMatchObject([
+        {
+          cmd: `${params.sdk} pub upgrade`,
+        },
+      ]);
+    });
+
+    it(`returns updated ${params.sdk} pubspec.lock for lockfile maintenance`, async () => {
+      const execSnapshots = mockExecAll();
+      fs.getSiblingFileName.mockReturnValueOnce(lockFile);
+      fs.readLocalFile.mockResolvedValueOnce(oldLockFileContent);
+      fs.readLocalFile.mockResolvedValueOnce(newLockFileContent);
+      expect(
+        await pub.updateArtifacts({
+          ...updateArtifact,
+          newPackageFileContent: params.packageFileContent,
+          config: { ...config, updateType: 'lockFileMaintenance' },
+        })
+      ).toEqual([
+        {
+          file: {
+            type: 'addition',
+            path: lockFile,
+            contents: newLockFileContent,
+          },
+        },
+      ]);
+      expect(execSnapshots).toMatchObject([
+        {
+          cmd: `${params.sdk} pub upgrade`,
+        },
+      ]);
+    });
+
+    it(`supports ${params.sdk} docker mode`, async () => {
+      GlobalConfig.set({ ...adminConfig, binarySource: 'docker' });
+      const execSnapshots = mockExecAll();
+      fs.getSiblingFileName.mockReturnValueOnce(lockFile);
+      fs.readLocalFile.mockResolvedValueOnce(oldLockFileContent);
+      fs.readLocalFile.mockResolvedValueOnce(newLockFileContent);
+      expect(
+        await pub.updateArtifacts({
+          ...updateArtifact,
+          newPackageFileContent: params.packageFileContent,
+        })
+      ).toEqual([
+        {
+          file: {
+            type: 'addition',
+            path: lockFile,
+            contents: newLockFileContent,
+          },
+        },
+      ]);
+      expect(execSnapshots).toMatchObject([
+        {
+          cmd: 'docker pull renovate/sidecar',
+        },
+        {
+          cmd: 'docker ps --filter name=renovate_sidecar -aq',
+        },
+        {
+          cmd:
+            'docker run --rm --name=renovate_sidecar --label=renovate_child ' +
+            '-v "/tmp/github/some/repo":"/tmp/github/some/repo" ' +
+            '-v "/tmp/cache":"/tmp/cache" ' +
+            '-e BUILDPACK_CACHE_DIR ' +
+            '-e CONTAINERBASE_CACHE_DIR ' +
+            '-w "/tmp/github/some/repo" ' +
+            'renovate/sidecar ' +
+            'bash -l -c ' +
+            `"install-tool ${params.sdk} 3.3.9 && ${params.sdk} pub upgrade"`,
```

Comment:
move the quotes to the previous part and also split commands. checkout other managers for samples